src/builtins.c

```diff
@@ -1409,7 +1411,7 @@ size_t jl_static_show_x(JL_STREAM *out, jl_value_t *v, int depth)
         n += jl_printf(out, ":%s", ((jl_sym_t*)v)->name);
     }
     else if (jl_is_gensym(v)) {
-        n += jl_printf(out, "GenSym(%d)", ((jl_gensym_t*)v)->id);
+        n += jl_printf(out, "GenSym(%d)", (int)((jl_gensym_t*)v)->id);
```

Comment:
There's no portable way to get a format string for `size_t`.

`PRIxPTR` could work with a cast to `intptr_t`.